### PR TITLE
Lovingly hand crafted proto

### PR DESCRIPTION
**Pull request by bhechinger**
This is the new layout for the esi-srv protobuf definitions. This is much more sane and will be much more readable once all the ESI endpoints get added. (Auto-generation created a 50k line go file. Yuck.)

### CODE REVIEW

File: proto/alliance.proto [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    rpc GetAllianceNames (AllianceNamesRequest) returns (AllianceNamesResponse) {
+    };
+}
+
+message AllianceRequest {
+    int32 AllianceId = 1;
+}
+
+message AllianceNamesRequest {
+    repeated int64 AllianceIds = 1;
```
Comment (maurerit) [inline proto/alliance.proto]: should be int32's?

File: proto/alliance.proto [hunk 2]
```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+
+package chremoas.esi;
+
+import "common.proto";
+
+service AllianceQuery {
+    rpc GetAlliances (EmptyRequest) returns (AlliancesResponse) {
+    };
+    rpc GetAllianceId (AllianceRequest) returns (AllianceIdResponse) {
```
Comment (maurerit) [inline proto/alliance.proto]: I would prefer (Get|Find)ByAllianceRequest with a response named `AllianceResponse`

File: proto/alliance.proto [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    };
+    rpc GetAllianceIdCorporations (AllianceRequest) returns (CorporationsResponse) {
```
Comment (maurerit) [inline proto/alliance.proto]: I think I would prefer (Get|Find)CorporationsByAllianceRequest

File: proto/search.proto [hunk 4]
```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package chremoas.esi;
+
+service SearchQuery {
+    rpc GetSearch (SearchRequest) returns (SearchResponse) {
```
Comment (maurerit) [inline proto/search.proto]: Something says I would like `Search` more than GetSearch.  I see no reason to call out the http method that happens on the backend of this internal call.  I don't think I'd hate this TOO much though ;)

File: proto/alliance.proto [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+message AllianceIdResponse {
+    Alliance Alliance = 1;
+}
+
+message AllianceIconResponse {
+    Icons Icons = 1;
+}
+
+message CorporationsResponse {
+    repeated int32 Corporations = 1;
```
Comment (maurerit) [inline proto/alliance.proto]: Please call this `CorporationIds` or just `Ids`

File: proto/alliance.proto [hunk 6]
```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+
+package chremoas.esi;
+
+import "common.proto";
+
+service AllianceQuery {
+    rpc GetAlliances (EmptyRequest) returns (AlliancesResponse) {
```
Comment (maurerit) [inline proto/alliance.proto]: Please call this response `AllianceIdsResponse`

File: proto/alliance.proto (hunk 1 above)
Comment (bhechinger) [reply proto/alliance.proto]: https://github.com/antihax/goesi/blob/master/esi/docs/AllianceApi.md#getalliancesnames says int64. that doesn't *feel* right, but I'm going to go with what the auto-generated docs say.

File: proto/search.proto (hunk 4 above)
Comment (bhechinger) [reply proto/search.proto]: Yeah, I'm down with that.

File: proto/alliance.proto (hunk 1 above)
Comment (maurerit) [reply proto/alliance.proto]: makes sense, just wanted to make sure this wasn't an error.

**Comment by bhechinger** [comment]
This is all the new stuff that we discussed on Slack. Please review.

**Pull request merged**